server/src/services/calculateAchievementsStatus.ts

```diff
@@ -0,0 +1,35 @@
+import { CalculateAchievementsStatus } from '../types/interfaces';
+
+export const calculateAchievementsStatus: CalculateAchievementsStatus = (
+  challengeId,
```

Comment:
I would suggest to pass here just a challenge as for now there is a lot of logic that seem a bit redundant